### DISCUSSION TITLE
Dialogic Signal: Can't return dictionary from on_Dialogue_event_start

### DISCUSSION THREAD

**Discussion by PuteraSinuraya** [dialogic-godot/dialogic#993] [Q&A]
Hello people. In my project, everytime a dialogue starts, a signal should be emitted to indicate that a particular character is talking. The event_start signal would pass "type" and "event". "type" would return a single string while "event" would return a... dictionary?

When I utilize
`event["character"]`
I got an error stating that event is not a dictionary but a string.
I really need to be able to return the person who is talking and the expression at any moment of time. I am desperately needing a solution.
[todrive.txt](https://github.com/coppolaemilio/dialogic/files/9079278/todrive.txt)

**Comment by zaknafean** [comment]
WIthout more information its hard to say whats going on. Can you print(type) before the crash happens so we can see which event is causing the issue? A screen shot of the code would also help.

**Reply by PuteraSinuraya** [reply]
Sure. Here is my code.
`extends Node2D

onready var mydialogue = Dialogic.start("first")
func _ready():
	add_child(mydialogue)

func _on_Dialog_event_start(type, event):
	print(event)
	print(typeof(event))
	print(event["character"])
	pass`
And attached below is my output screenshot and the debugger.
![image](https://user-images.githubusercontent.com/60869659/178188513-7903af54-b399-4f58-8fd6-c9a7f975845c.png)
![image](https://user-images.githubusercontent.com/60869659/178188596-fa67b5d3-ef7a-4055-ac08-3ca64e38f27c.png)


If I commented out the `print(event['character']`, It would not crash and I get this output.
`func _on_Dialog_event_start(type, event):
	print(event)
	print(typeof(event))
	#print(event["character"])`
![image](https://user-images.githubusercontent.com/60869659/178188914-2e2695f8-4bb1-4ee3-856b-62a88eda1753.png)

I am not sure why `timeline-1657460404.json and 4` are printed though

In any case, I need to be able to return the name of the speaking character at any point of time from `event["character"]` .
